Feature/quality control

ragga3 opened the pull request:
Feat(Setup) : Monorepo for code quality control packages.

Commit:
feat(Code Quality Control packages for javascript , typescript , scss , styled component): added pac

affects: @xt-pagesource/eslint-config, @xt-pagesource/stylelint-config

Code Quality Control packages for javascript , typescript , scss , styled component

Commit:
perf(): node module folder deleted

Commit:
docs(Configuration package documentation added): how to install packages and usages

affects: @xt-pagesource/stylelint-config

Configuration package documentation added on how to install packages and usages

Commit:
refactor(git hooks added): git hooks added

git hooks and added configuration husky and lint stageds

ankurkakroo2 commented on packages/eslint-config/index.js:
```diff
@@ -0,0 +1,56 @@
+module.exports = {
+  parser: '@typescript-eslint/parser',
```
We need to revisit this whole config. 
This is a typescript parser which is something we are not supporting.

ragga3 replied:
@ankurkakroo2  done

ankurkakroo2 commented on packages/eslint-config/index.js:
```diff
@@ -0,0 +1,56 @@
+    window: true,
+    React: 'writable',
+  },
+  plugins: [
```
Lets make the config smaller. We would not want to enforce a ton of stuff. Let the users extend to whatever they need. Also, lets keep the defaults of the rules we are using

ragga3 replied:
@ankurkakroo2  do we need to provide very basic configuration? initially I thought to provide basic version but thought than what is intent of package ? Please confirm.

ankurkakroo2 replied:
We are already consuming airbnb and other rules which are massive in itself. I am not saying write as less as possible but rather, use only those configs which we are sure that the user should inherit.

The primary intention for these kind of packages is usually consistency among multiple teams.  We have learnt this before as well. The more we put, the harder it is for us.

Lets try to provide a just sufficient extensible config.

vinodloha replied:
Starting small is a good idea and carefully adding curated items to make it better

ankurkakroo2 commented on packages/eslint-config/package.json:
```diff
@@ -0,0 +1,36 @@
+    "prettier": "^2.0.5",
+    "@typescript-eslint/eslint-plugin": "^2.31.0",
+    "@typescript-eslint/parser": "^2.31.0",
+    "typescript": "^3.1.3"
```
Why typescript ?

ragga3 replied:
@ankurkakroo2  done

ankurkakroo2 commented on packages/eslint-config/package.json:
```diff
@@ -0,0 +1,36 @@
+  "files": [
+    "index.js"
+  ],
+  "dependencies": {
```
We need to keep them in peer dependencies.
Think about it - the execution will happen where the actual code will be and where the module will be consumed. 

It will be working in your examples because node dependency resolution work bottom up. If you move your example outside the folder, it may not work

ragga3 replied:
@ankurkakroo2  done

ankurkakroo2 commented on packages/stylint-config/example/src/index.js:
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Lets keep this examples outside of the packages and at root.

ragga3 replied:
@ankurkakroo2  My thought to keep it at package specific only,

ankurkakroo2 replied:
Sure. Lets talk about it. I suggested this because you consumed both configs in your example.
Also, if we move it out, we can demonstrate everything in 1 example (considering only react )

ragga3 replied:
@ankurkakroo2  As discussed move to package folder

ankurkakroo2 commented on packages/eslint-config/package.json:
```diff
@@ -0,0 +1,36 @@
+  ],
+  "dependencies": {
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb": "^18.1.0",
```
We also need to find a way to add all dependencies automatically to the consumer of this package. 

Eg: I install this package in my app, the needed dependencies and scripts will be automatically added to my package json.

Lets talk about it if you need any help.

ankurkakroo2 commented on lerna.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "packages": ["packages/*"],
```
If we are providing react specific quality control. Lets structure our packages in a slightly different manner.

Proposed structure:

packages
 -> react ->  javascript  -> eslint-config
      

Lerna.json

`
{
  "packages": [
          "packages/react/javascript/eslint-config", 
          "packages/react/javascript/styleline-config"
   ]
}
`

OR we can change the repo name to react-quality-control or react-standards :)

ragga3 replied:
@ankutr that was not the ask and intent to create react specific. user can also turned off any rules or update rules if required. Why do you think its react specific only?

ankurkakroo2 replied:
Yeah. Thats why I mentioned an OR condition. :) If the name of the repo is generic, the code and the relevant structure should support additions

ragga3 replied:
@ankurkakroo2  You wil be discussing this approach with vinod and confirm.

vinodloha replied:
@ankurkakroo2 I like your idea, we could use the same setup for our future Vue work or any thing else in future. 

My thought was to create that differentiation in package name itself, like @xt-pagesource/react-next-eslint

ankurkakroo2 commented on lerna.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "packages": ["packages/*"],
+  "npmClient": "yarn",
+  "useWorkspaces": true,
```
any specific reason to use Workspaces ?

ragga3 replied:
done

ankurkakroo2 commented on package.json:
```diff
@@ -0,0 +1,41 @@
+    "commit": "npx git-cz"
+  },
+  "devDependencies": {
+    "@commitlint/cli": "^7.0.0",
```
We also need to think about creating packages for husky, prettier and commitlint.
That can be done on top of this later

ragga3 replied:
@ankurkakroo2  yes we can add for husky not sure about prettier really need separate package?

vinodloha replied:
Yes, we will need for husky, prettier and commitlint.

ragga3 replied:
@vinodloha  @ankurkakroo2  We can add more as much as packages to quality control but this comments should not be blocker for PR merge.

ankurkakroo2 commented on packages/eslint-config/package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "@xt-pagesource/eslint-config",
```
we need to change the names of the packages if we change the structure.

Because we may want to use the same repo for vanilla js, angular, vue etc. The package name should clearly signify the same

eg: xt-pagesource/eslint-config-react

ragga3 replied:
@ankurkakroo2  - Please have discussion with Vinod that was not the intent.

ankurkakroo2 replied:
ok.

@vinodloha : Please have a look at https://github.com/pagesource/quality-control/pull/1#discussion_r422138601 as well.

We can talk about it

ragga3 replied:
@ankurkakroo2 - I mean to say that was not the intent and ask to create different packages for each lib/framework.

ragga3 replied:
this comments is same as above comment for folder structure changes and move to react specific.

ankurkakroo2 commented on packages/stylint-config/index.js:
```diff
@@ -0,0 +1,27 @@
+
+module.exports = {
+  processors: ['stylelint-processor-styled-components'],
+  plugins: ['stylelint-scss'],
```
we are not using scss anywhere

ankurkakroo2 commented on packages/stylint-config/package.json:
```diff
@@ -0,0 +1,29 @@
+  "files": [
+    "index.js"
+  ],
+  "dependencies": {
```
Most comments on eslint page should be applicable here as well

Commit:
refactor(code review comments incorporation): remove typescript support

affects: @xt-pagesource/eslint-config, sample-app, @xt-pagesource/stylelint-config

remove typescript support, workspace set to false , dependencies move to peer, remove scss support
from stylelint

ghost commented:
@ankurkakroo2 peer review comments incorporated please review except the one which we need to discuss.

Commit:
husky config test

Commit:
feat: Added more packages to quality control

Commit:
feat: updated packages

Commit:
perf: config rules updated

Commit:
feat: peer depedency auto feature instatlled

ghost commented:
@vinodloha  @ankurkakroo2  PR updated please review

ghost commented:
@ankurkakroo2  Please review this PR if needed any fix do let me know. I will pick up.

@vinodloha - FYI

Commit:
feat(commitlint): package for commitlint added

Package for commitlint added.
Rules of eslint added.

Commit:
fix(eslint): eslint package updated

eslint package will create .eslintrc.js file in the folder.
Added some rules for eslint.

Commit:
fix(eslint): eslint file being generated in the folder

Fixed the eslint postinstall file.

Commit:
feature(semantic release): semantic release feature created

Added feature for semantic release.

Commit:
feature(eslint-config-next): eslint-config-next package created

Eslint-config-next package created and dependency added.

vinodloha committed:
Merge pull request #3 from pagesource/feature/eslint

Feature/eslint

Commit:
fix(eslint-config-next): added feature to auto create file .eslintrc.js

Added feature to auto create file .eslintrc.js
Added documentation for the package.

vinodloha committed:
Merge pull request #4 from pagesource/feature/semantic-release

feature(semantic release): semantic release feature created

vinodloha committed:
Merge pull request #5 from pagesource/feature/eslint-config-next

Feature/eslint config next
